src/lcm/LifeCycleStates.cc

```diff
@@ -2496,3 +2673,500 @@ void LifeCycleManager::trigger_resize_failure(int vid)
+        int keep_last      = backups.keep_last();
+        Backups::Mode mode = backups.mode();
+
+        long long reserved_sz = vm->backup_size(ds_deltas);
+        long long real_sz     = 0;
+
+        one_util::str_cast(backups.last_backup_size(), real_sz);
+
+        ds_deltas.add("DATASTORE", backups.last_datastore_id());
+
+        if (mode == Backups::FULL || incremental_id == -1)
+        {
+            ds_deltas.add("IMAGES", 1);
+        }
+        else
+        {
+            ds_deltas.add("IMAGES", 0);
+        }
+
+        switch(vm->get_lcm_state())
+        {
+            case VirtualMachine::BACKUP:
+                vm->set_state(VirtualMachine::RUNNING);
+                break;
+
+            case VirtualMachine::BACKUP_POWEROFF:
+                vm->set_state(VirtualMachine::POWEROFF);
+                vm->set_state(VirtualMachine::LCM_INIT);
+                break;
+
+            default:
+                vm->log("LCM", Log::ERROR, "backup_success, VM in wrong state");
+                vm.reset();
+
+                Quotas::ds_del(vm_uid, vm_gid, &ds_deltas);
+                return;
+        }
+
+        /* ------------------------------------------------------------------ */
+        /* Create Backup image if needed                                      */
+        /* ------------------------------------------------------------------ */
+        int image_id   = -1;
+        int increments = -1;
+
+        std::set<int> delete_ids;
+
+        if (mode == Backups::FULL || incremental_id == -1)
+        {
+            image_id = create_backup_image(vm.get(), error);
+
+            if ( image_id == -1 )
+            {
+                goto error_image_create;
+            }
+
+            backups.add(image_id);
+
+            backups.remove_last(delete_ids);
+
+            if (auto image = ipool->get(image_id))
+            {
+                vector<int> backup_disk_ids;
+
+                vm->get_disks().backup_disk_ids(backups.do_volatile(), backup_disk_ids);
+
+                for (auto id : backup_disk_ids)
+                {
+                    image->add_backup_disk(id);
+                }
+
+                ipool->update(image.get());
+            }
+        }
+
+        /* ------------------------------------------------------------------ */
+        /* Update backup information for increments                           */
+        /* ------------------------------------------------------------------ */
+
+        if (mode == Backups::INCREMENT)
+        {
+            Increment::Type itype;
+
+            if (incremental_id == -1)
+            {
+                backups.incremental_backup_id(image_id);
+                itype = Increment::FULL;
+            }
+            else
+            {
+                image_id = backups.incremental_backup_id();
+                itype = Increment::INCREMENT;
+            }
+
+            if (auto image = ipool->get(image_id))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This snippet may contain an accidental use of incorrect operation. Make sure that the operation being performed is in fact what you intended.